spec/unit/entities/service_spec.rb

```diff
@@ -68,6 +70,73 @@
     end
   end
 
+  context 'Service.find' do
+    let(:system_name) { random_lowercase_name }
+    let(:service_info) { { remote: remote, ref: system_name } }
+
+    it 'an exception is raised when remote is not configured' do
+      expect(remote).to receive(:show_service).and_raise(StandardError)
+      expect do
+        described_class.find(service_info)
+      end.to raise_error(StandardError)
+    end
+
+    it 'returns nil when the service does not exist' do
+      expect(remote).to receive(:show_service).and_raise(ThreeScale::API::HttpClient::NotFoundError)
+      expect(remote).to receive(:list_services).and_return([{"system_name" => "sysname1"}, {"system_name" => "sysname2"}])
+      expect(described_class.find(service_info)).to be_nil
+    end
+
+    it 'service instance is returned when specifying an existing service ID' do
+      expect(remote).to receive(:show_service).and_return({"id" => system_name, "system_name" => "sysname1"})
+      expect(remote).not_to receive(:list_services)
```

Comment:
without this negative expectation, the test would still fail when `list_services` are called, because remote is `double`

Reply:
:+1: I'll remove it